docs/spec/generics.rst

```diff
@@ -496,11 +496,10 @@ be a subtype of the boundary type. Example::
   longer({1}, {1, 2})  # ok, return type set[int]
   longer([1], {1, 2})  # ok, return type a supertype of list[int] and set[int]
 
-An upper bound cannot be combined with type constraints (as used in
-``AnyStr``, see the example earlier); type constraints cause the
-inferred type to be *exactly* one of the constraint types, while an
-upper bound just requires that the actual type is a subtype of the
-boundary type.
+An upper bound cannot be combined with type constraints (as used in ``AnyStr``,
+see the example earlier); type constraints cause the inferred type to be
```

Comment:
Off-topic (this should be tackled separately), but we shouldn't use `AnyStr` as an example in this document. It's deprecated on py313+ because it has a truly terrible name (it neither means "any string", nor has anything to do with `Any`), and scheduled for removal in 3.18. See https://github.com/python/cpython/issues/105578